cmd/client.go

```diff
@@ -17,23 +19,75 @@ type Client struct {
 	mergeId        int
 	gitlabInstance string
 	authToken      string
+	logPath        string
+	debug          bool
 	git            *gitlab.Client
 }
 
-type Logger struct {
-	Active bool
+var requestLogger retryablehttp.RequestLogHook = func(l retryablehttp.Logger, r *http.Request, i int) {
+	logPath := os.Args[len(os.Args)-1]
+
+	file, err := os.OpenFile(logPath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	clonedRequest := r.Clone(r.Context())
+	var data []byte
+	reader := bufio.NewReader(clonedRequest.Body)
+	_, err = reader.Read(data)
+	if err != nil {
+		panic(err)
+	}
+
+	clonedRequest.Header.Set("Private-Token", "xxxx")
+	output := map[string]interface{}{
+		"Method":  clonedRequest.Method,
+		"URL":     clonedRequest.URL.String(),
+		"Body":    string(data),
+		"Headers": clonedRequest.Header,
+	}
+
+	logString, err := json.MarshalIndent(output, "", "  ")
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = file.Write([]byte("-- REQUEST: \n"))
+	_, err = file.Write(logString)
+	_, err = file.Write([]byte("\n"))
 }
 
-func (l Logger) Printf(s string, args ...interface{}) {
-	logString := fmt.Sprintf(s+"\n", args...)
+var responseLogger retryablehttp.ResponseLogHook = func(l retryablehttp.Logger, response *http.Response) {
 	logPath := os.Args[len(os.Args)-1]
 
 	file, err := os.OpenFile(logPath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
 	if err != nil {
 		panic(err)
 	}
 	defer file.Close()
-	_, err = file.Write([]byte(logString))
+
+	var data []byte
+	reader := bufio.NewReader(response.Body)
+	_, err = reader.Read(data)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
This is similar to previous comment.
```suggestion
	var data []byte
	if response.Body != nil {
		data, err = io.ReadAll(response.Body)
		if err != nil {
			panic(err)
		}
		response.Body = io.NopCloser(bytes.NewBuffer(data))
	}
```
Here we cannot leave response.Body empty ( already read ) otherwise handler will not be able to get content and will return error.

There is also alternative in httputil package https://pkg.go.dev/net/http/httputil#DumpResponse